improver_tests/calibration/reliability_calibration/test_ApplyReliabilityCalibration.py

```diff
@@ -227,23 +227,89 @@ def test_exception_without_relative_to_threshold(self):
 
         msg = "Cube threshold coordinate does not define whether"
         with self.assertRaisesRegex(ValueError, msg):
-            self.plugin._ensure_monotonicity(self.forecast)
+            self.plugin._ensure_monotonicity_across_thresholds(self.forecast)
+
+
+class Test__combine_bin_pair_for_observation_frequency_monotonicity(unittest.TestCase):
+
+    """Test the _combine_bin_pair_for_observation_frequency_monotonicity method"""
+
+    def setUp(self):
+        """Set up forecast count"""
+        self.forecast_count = np.array([1000, 1000, 1000, 1000, 1000], dtype=np.float32)
+        self.forecast_probability_sum = np.array(
+            [0, 250, 500, 750, 1000], dtype=np.float32
+        )
+
+        self.expected_non_monotonic = np.array(
```

Comment:
I don't love this name as the results are clearly monotonic.
```expected_enforced_monotonic```?

Reply:
I've modified this as suggested.